libdnf/transaction/rpm_package.cpp

```diff
@@ -0,0 +1,199 @@
+
+
+uint32_t Package::get_epoch_int() const {
+    if (get_epoch().empty()) {
+        return 0;
+    }
+    return std::stoi(get_epoch());
+}
+
+
+std::string Package::to_string() const {
+    return libdnf::rpm::to_nevra_string(*this);
+}
+
+
+
+/*
+TransactionItemReason Package::resolveTransactionItemReason(
+    libdnf::utils::SQLite3 & conn,
+    const std::string & name,
+    const std::string & arch,
+    [[maybe_unused]] int64_t maxTransactionId) {
+    const char * sql = R"**(
+        SELECT
+            ti.action as action,
+            ti.reason as reason
+        FROM
+            trans_item ti
+        JOIN
+            trans t ON ti.trans_id = t.id
+        JOIN
+            rpm i USING (item_id)
+        WHERE
+            t.state = 1
+            AND ti.action not in (3, 5, 7, 10)
+            AND i.name = ?
+            AND i.arch = ?
+        ORDER BY
+            ti.trans_id DESC
+        LIMIT 1
+    )**";
+
+    if (arch != "") {
+        libdnf::utils::SQLite3::Query query(conn, sql);
+        query.bindv(name, arch);
+
+        if (query.step() == libdnf::utils::SQLite3::Statement::StepResult::ROW) {
+            auto action = static_cast<TransactionItemAction>(query.get<int64_t>("action"));
+            if (action == TransactionItemAction::REMOVE) {
+                return TransactionItemReason::UNKNOWN;
+            }
+            auto reason = static_cast<TransactionItemReason>(query.get<int64_t>("reason"));
+            return reason;
+        }
+    } else {
+        const char * arch_sql = R"**(
+            SELECT DISTINCT
+                arch
+            FROM
+                rpm
+            WHERE
+                name = ?
+        )**";
+
+        libdnf::utils::SQLite3::Query arch_query(conn, arch_sql);
+        arch_query.bindv(name);
+
+        TransactionItemReason result = TransactionItemReason::UNKNOWN;
+
+        while (arch_query.step() == libdnf::utils::SQLite3::Statement::StepResult::ROW) {
+            auto rpm_arch = arch_query.get<std::string>("arch");
+
+            libdnf::utils::SQLite3::Query query(conn, sql);
+            query.bindv(name, rpm_arch);
+            while (query.step() == libdnf::utils::SQLite3::Statement::StepResult::ROW) {
+                auto action = static_cast<TransactionItemAction>(query.get<int64_t>("action"));
+                if (action == TransactionItemAction::REMOVE) {
+                    continue;
+                }
+                auto reason = static_cast<TransactionItemReason>(query.get<int64_t>("reason"));
+                if (reason > result) {
+                    result = reason;
+                }
+            }
+        }
+        return result;
+    }
+    return TransactionItemReason::UNKNOWN;
+}
+*/
+
+/**
+ * Compare RPM packages
+ * This method doesn't care about compare package names
+ * \param other Package to compare with
+ * \return true if other package is newer (has higher version and/or epoch)
+ */
```

Comment:
Incorrect format of dock string => `///`.

Reply:
I've removed the doc string completely.
It belongs to hpp and also such operators do not need any explanation.
There's already TODO in the code to improve it.